base/nullable.jl

```diff
@@ -3,7 +3,7 @@
 immutable NullException <: Exception
 end
 
-Nullable{T}(value::T, isnull::Bool=false) = Nullable{T}(value, isnull)
+Nullable{T}(value::T, hasvalue::Bool=true) = Nullable{T}(value, hasvalue)
```

Comment:
Yes. We should document it in such a way that it's clear that you have more freedom than you usually need and that its use is a performance optimization.